### PR TITLE
fix: stroke and fills not updating when you change property

### PR DESCRIPTION
**Pull request by Zyie**
Fixes: #10426

we now proxy the fill and stroke to make sure we can capture property changes

### CODE REVIEW

File: src/scene/text/TextStyle.ts [hunk 1]
```diff
@@ -535,6 +560,13 @@ export class TextStyle extends EventEmitter<{
             }
         });
     }
+
+    private _isFillStyle(value: FillInput): value is FillStyle
+    {
+        // eslint-disable-next-line no-eq-null, eqeqeq
+        return (value != null
```
Comment (bigtimebuddy) [inline src/scene/text/TextStyle.ts:567]: Why violate the lint here?

**Review by bigtimebuddy — APPROVED** [review]

File: src/scene/text/TextStyle.ts [hunk 2]
```diff
@@ -416,6 +429,19 @@ export class TextStyle extends EventEmitter<{
         if (value === this._originalStroke) return;
 
         this._originalStroke = value;
+
+        // eslint-disable-next-line no-eq-null, eqeqeq
```
Comment (bigtimebuddy) [inline src/scene/text/TextStyle.ts:433]: What this old? Lint exception doesn't make sense to me here.
```suggestion
```

File: src/scene/text/TextStyle.ts (hunk 1 above)
Comment (bigtimebuddy) [reply src/scene/text/TextStyle.ts:567]: I think being verbose here is okay. Style should only be over ridden as a necessity. You could try using the coalesce operator.

```ts
(value ?? null) !== null
```

**Review by GoodBoyDigital — APPROVED** [review]
very cool! nice work!

**Pull request merged**